pmd-plsql/etc/grammar/PLSQL.jjt

```diff
@@ -7058,3 +7064,14 @@ ASTJavaInterfaceClass JavaInterfaceClass(): {}
         )
 	{ jjtThis.setImage(token.getImage()) ; jjtThis.value = token ; return jjtThis ; }
 }
+
+/**
+ * https://docs.oracle.com/en/database/oracle/oracle-database/23/lnpls/SQL_MACRO-clause.html#LNPLS-GUID-292C3A17-2A4B-4EFB-AD38-68DF6380E5F7
+ */
+ASTSqlMacroClause SqlMacroClause(): {}
+{
+       (
+         <SQL_MACRO> [ <LPAREN> [ <TYPE> "=>" ] ( <SCALAR> | <TABLE> ) { jjtThis.setType(token.getImage()); } <RPAREN> ]
```

Comment:
```suggestion
         "SQL_MACRO" [ <LPAREN> [ <TYPE> "=>" ] ( "SCALAR" | <TABLE> ) { jjtThis.setType(token.getImage()); } <RPAREN> ]
```

Reply:
I remembered, it should be

```suggestion
        LOOKAHEAD({isKeyword("SQL_MACRO")}) KEYWORD("SQL_MACRO") [ <LPAREN> [ <TYPE> "=>" ] ( LOOKAHEAD({isKeyword("SCALAR")}) KEYWORD("SCALAR") | <TABLE> ) { jjtThis.setType(token.getImage()); } <RPAREN> ]
```

That way, we don't need to define a token at all and parse "SQL_MACRO" as an identifier.